### ISSUE TITLE
采用cityscape 训练2Dunet，在3090上，精度达不到文中给出的精度

### ISSUE THREAD

**Issue by HongweiChe** [PaddlePaddle/PaddleSeg#3510]
文中给出的：对应 65%
https://github.com/PaddlePaddle/PaddleSeg/tree/develop/configs/unet

Model | Backbone | Resolution | Training Iters | Batch Size | mIoU | mIoU (flip) | mIoU (ms+flip) | Links
-- | -- | -- | -- | -- | -- | -- | -- | --
UNet | - | 1024x512 | 160000 | 4 | 65.00% | 66.02% | 67.29% | model \| log \| vdl

实际在3090跑出的结果 "best_mean_iou": 0.6177095346989564,   **精度相差2-3个点**
具体config如下：
---------------Config Information---------------
batch_size: 4
iters: 160000
train_dataset:
  dataset_root: /home/devdata/dataset/cityscapes
  mode: train
  transforms:
  - max_scale_factor: 2.0
    min_scale_factor: 0.5
    scale_step_size: 0.25
    type: ResizeStepScaling
  - crop_size:
    - 1024
    - 512
    type: RandomPaddingCrop
  - type: RandomHorizontalFlip
  - brightness_range: 0.4
    contrast_range: 0.4
    saturation_range: 0.4
    type: RandomDistort
  - type: Normalize
  type: Cityscapes

val_dataset:
  dataset_root: /home/devdata/dataset/cityscapes
  mode: val
  transforms:
  - type: Normalize
  type: Cityscapes
optimizer:
  momentum: 0.9
  type: SGD
  weight_decay: 4.0e-05
lr_scheduler:
  end_lr: 0
  learning_rate: 0.01
  power: 0.9
  type: PolynomialDecay
loss:
  coef:
  - 1
  types:
  - type: CrossEntropyLoss
model:
  num_classes: 19
  pretrained: null
  type: UNet
  use_deconv: false
to_static_training: true
------------------------------------------------

针对链接里面的config对比参数没有什么差别， 
仅仅有区别：
1.  lr_scheduler:  type: PolynomialDecay
2. to_static_training: true  
理论这些应该都不影响精度，求大佬指点~

**Comment by shiyutang** [comment]
你好，to_static_training: true是转为静态图训练，另外学习率差异也影响精度，二者都可能会导致精度差异的，建议不要修改代码和配置文件重新训练～

**Comment by HongweiChe** [comment]
用过原始参数，跑出来的精度比61还要低~

**Comment by onecatcn** [comment]
@HongweiChe 请问使用的paddle版本和cuda版本能否提供？

**Comment by HongweiChe** [comment]
paddlepaddle-gpu        2.5.1.post120
NVIDIA-SMI 525.60.11    Driver Version: 525.60.11    CUDA Version: 12.0

**Comment by HongweiChe** [comment]
2023-09-29 17:31:14 [INFO]      [TRAIN] epoch: 861, iter: 159999/160000, loss: 0.0794, lr: 0.000000, batch_cost: 0.6411, reader_cost: 0.00024, ips: 6.2397 samples/sec | ETA 00:00:00
2023-09-29 17:31:14 [INFO]      [TRAIN] epoch: 861, iter: 160000/160000, loss: 0.1049, lr: 0.000000, batch_cost: 0.7128, reader_cost: 0.00029, ips: 5.6116 samples/sec | ETA 00:00:00
2023-09-29 17:31:14 [INFO]      Start evaluating (total_samples: 500, total_iters: 125)...
125/125 - 18s - batch_cost: 0.1479 - reader cost: 0.0104
2023-09-29 17:31:33 [INFO]      [EVAL] #Images: 500 mIoU: 0.6377 Acc: 0.9425 Kappa: 0.9253 Dice: 0.7547
2023-09-29 17:31:33 [INFO]      [EVAL] Class IoU:
[0.9673 0.77   0.8947 0.3533 0.4561 0.5838 0.6357 0.7295 0.9144 0.5799
 0.9346 0.7447 0.4243 0.9146 0.3136 0.5231 0.24   0.4122 0.7234]
2023-09-29 17:31:33 [INFO]      [EVAL] Class Precision:
[0.9805 0.8785 0.9338 0.6571 0.7093 0.7967 0.7924 0.8867 0.953  0.7831
 0.9565 0.8214 0.691  0.9466 0.5411 0.7338 0.5266 0.6391 0.8309]
2023-09-29 17:31:33 [INFO]      [EVAL] Class Recall:
[0.9862 0.8618 0.9553 0.4332 0.561  0.686  0.7628 0.8046 0.9575 0.6909
 0.9761 0.8886 0.5237 0.9644 0.4272 0.6457 0.3061 0.5372 0.8483]
**2023-09-29 17:31:33 [INFO]      [EVAL] The model with the best validation mIoU (0.6377) was saved at iter 160000.**
<class 'paddle.nn.layer.conv.Conv2D'>'s flops has been counted
Customize Function has been applied to <class 'paddle.nn.layer.norm.SyncBatchNorm'>
<class 'paddle.nn.layer.activation.ReLU'>'s flops has been counted
Cannot find suitable count function for <class 'paddle.nn.layer.pooling.MaxPool2D'>. Treat it as zero FLOPs.
Total Flops: 253750673408     Total Params: 13414163
I0929 17:31:35.444276   214 tcp_store.cc:273] receive shutdown event and so quit from MasterDaemon run loop
LAUNCH INFO 2023-09-29 17:31:37,963 Pod completed
LAUNCH INFO 2023-09-29 17:31:37,967 Exit code 0
Type  :qa  and press <Enter> to exit Vim

**Comment by HongweiChe** [comment]
V100 上    NVIDIA-SMI 450.203.03   Driver Version: 450.203.03   CUDA Version: 11.0  跑出的上面结果，依旧没有达到65%，原始代码没有做任何改动

**Comment by shiyutang** [comment]
是使用4卡训练的么？源代码使用4卡训练。
![image](https://github.com/PaddlePaddle/PaddleSeg/assets/34859558/d0b52e52-249e-4ecc-8753-1643250b8209)

**Comment by HongweiChe** [comment]
是的 用的是4卡
------------Environment Information-------------
platform: Linux-5.4.0-128-generic-x86_64-with-glibc2.27
Python: 3.8.16 (default, Dec  7 2022, 01:12:13) [GCC 7.5.0]
Paddle compiled with cuda: True
NVCC: Not Available
cudnn: 7.6
GPUs used: 4
CUDA_VISIBLE_DEVICES: 0,1,2,3
GPU: ['GPU 0: Tesla V100-SXM2-32GB', 'GPU 1: Tesla V100-SXM2-32GB', 'GPU 2: Tesla V100-SXM2-32GB', 'GPU 3: Tesla V100-SXM2-32GB', 'GPU 4: Tesla V100-SXM2-32GB', 'GPU 5: Tesla V100-SXM2-32GB', 'GPU 6: Tesla V100-SXM2-32GB', 'GPU 7: Tesla V100-SXM2-32GB']
GCC: gcc (GCC) 8.2.0
PaddleSeg: 2.8.0
PaddlePaddle: 2.5.1
OpenCV: 4.5.5
------------------------------------------------
2023-09-28 09:34:42 [INFO]
---------------Config Information---------------
batch_size: 4
iters: 160000
train_dataset:
  dataset_root: /home/devdata/dataset/cityscapes
  mode: train
  transforms:
  - max_scale_factor: 2.0
    min_scale_factor: 0.5
    scale_step_size: 0.25
    type: ResizeStepScaling
  - crop_size:
    - 1024
    - 512
    type: RandomPaddingCrop
  - type: RandomHorizontalFlip
  - brightness_range: 0.4
    contrast_range: 0.4
    saturation_range: 0.4
    type: RandomDistort
  - type: Normalize
  type: Cityscapes
val_dataset:
  dataset_root: /home/devdata/dataset/cityscapes
  mode: val
  transforms:
  - type: Normalize
  type: Cityscapes
optimizer:
  momentum: 0.9
  type: SGD
  weight_decay: 4.0e-05
lr_scheduler:
  end_lr: 0
  learning_rate: 0.01
  power: 0.9
  type: PolynomialDecay
loss:
  coef:
  - 1
  types:
  - type: CrossEntropyLoss
model:
  num_classes: 19
  pretrained: null
  type: UNet
  use_deconv: false
------------------------------------------------

**Comment by shiyutang** [comment]
请给出进一步的训练日志，查看学习率、损失是否有差异

**Comment by HongweiChe** [comment]
[log_gpu_bs_4_eager.log](https://github.com/PaddlePaddle/PaddleSeg/files/12876674/log_gpu_bs_4_eager.log)

**Comment by HongweiChe** [comment]
我已经上传了训练时的log, 其中部分打印删除。
”查看学习率、损失是否有差异“   这个问题是不存在的，我们用的是原始代码没做任何修改跑出来的结果

**Comment by TingquanGao** [comment]
Thanks for this issue. As it has been inactive for a long time, we would close it. If you has any questions, please feel free to reopen or new issue, and we will follow up and resolve it.
